### PR TITLE
comments are not serialized with custom node input nodes

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

@andydandy74 reported missing comments in his converted clockwork package. 
~This PR serializes type parameter comments into the nodeView of input(symbol) nodes, using as simple as a converter as I could write.~

~we use the new converter when saving the workspaceViewModel.~

~At load time, we pull the comment out of the nodeView if the node we're loading is an input node, and then we construct a new type parameter using that comment and existing type paramter, and then set the inputSymbol equal to the string representation of that new type parameter.~

This PR serializes the `typedParameter.Summary` as Description using the existing `TypedParamterConverter` and adds some tests. It appears that if we load an existing custom node with no Description in the parameter we get an emptyString back from the symbol node - seems reasonable to me.

comments are always appended to the start of the input symbol and they are always wrapped with block comments as the parser throws away everything but the comment's string value. - ie

for `//a comment` we can only serialize `a comment`

sample serialization:
```
 "Parameter": {
        "Name": "inputName",
        "TypeName": "string",
        "TypeRank": 0,
        "DefaultValue": "\"a def string\"",
        "Description": " commentsssss"
      },
```

TODO:
- [x] This PR needs tests.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang 
@ramramps 
### FYIs

@smangarole @andydandy74

### CODE REVIEW

**Comment by ramramps** [comment]
@mjkkirschner Does `summary` serializes in `NodeView.Parameter`?  or as `Summary` in `NodeView`?

File: src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs [hunk 1]
```diff
@@ -26,6 +26,7 @@
 using Dynamo.Engine;
 using Dynamo.Scheduler;
 using Newtonsoft.Json.Linq;
+using Dynamo.Library;
```
Comment (QilongTang) [inline src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs]: Added this for `Nodes.CustomNodes.Symbol` type check?

**Comment by ramramps** [comment]
@mjkkirschner is there a reason for not adding this to `Parameter` block?
 "Parameter": {
        "Name": "var1",
        "TypeName": "System.DateTime",
        "TypeRank": 0,
        "DefaultValue": null
      },
"

File: src/DynamoCoreWpf/ViewModels/Core/Converters/SerializationConverters.cs [hunk 2]
```diff
@@ -78,6 +81,32 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
...
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var directSerialization = JsonConvert.SerializeObject(value);
+            var jobject = JObject.Parse(directSerialization);
+            //if the nodeView's model is a symbol node add some extra data.
+            if ((value as NodeViewModel).NodeModel is Symbol)
+            {
+                jobject.Add("Summary", ((value as NodeViewModel).NodeModel as Symbol).Parameter.Summary);
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/Converters/SerializationConverters.cs]: Can you check if empty summary will be kept and serialized as empty string? Just curious

**Comment by gregmarr** [comment]
It's better than adding it to the TypeName field as a comment, which is what this looks like it is doing.  If we add Parameter.Description it matches Port.Description.  I guess the alternative would be adding it under View.NodeModel.

**Comment by mjkkirschner** [comment]
@gregmarr this PR is meant to go into a release quite soon, not just master.  Does that cause any issues.

Summary already exists on paramter so it's very simple AFAIK to just serialize summary into the parameter...

I'm not sure what you mean here
`It's better than adding it to the TypeName field as a comment, which is what this looks like it is doing. ` We do a funny thing when setting the inputSymbol, the string is parsed back to a new TypedParamter...

**Comment by mjkkirschner** [comment]
right, so we would get:
```
"Parameter": {
"Name": "var1",
"TypeName": "System.DateTime",
"TypeRank": 0,
"DefaultValue": null,
"Summary" : "a comment"
},
```

@ramramps @QilongTang are we okay with that?

**Comment by mjkkirschner** [comment]
@gregmarr would this change to the schema make it into production?

**Comment by gregmarr** [comment]
Can we use `Description` to match `Port`, `Graph`, `InputData` and `OutputData`?

**Comment by QilongTang** [comment]
@mjkkirschner I am fine with it given that it is much cleaner, I just thought we do not want to implement it that way in the first place because this looks like a schema change

**Comment by mjkkirschner** [comment]
Yep, well I was trying to avoid them because I believe they cause work for @gregmarr and we have to reason about what will be in production at what time, but If @gregmarr is fine with it I think its cleaner.

**Comment by gregmarr** [comment]
We're generally okay with additions to the schema.  Since it's adding data that isn't otherwise there, it won't hurt anything.  We can have this in CoGS dev immediately, staging and production fairly soon.

**Comment by gregmarr** [comment]
Well, I'm the one who didn't include it in the schema in the first place, so I'm fine with it.  :)

**Comment by mjkkirschner** [comment]
@ramramps @gregmarr @QilongTang I will change this implementation to serialize like this:
```
"Parameter": {
"Name": "var1",
"TypeName": "System.DateTime",
"TypeRank": 0,
"DefaultValue": null,
"Description" : "a comment"
},```

**Comment by ramramps** [comment]
"Description" = "Summary" ?

**Comment by ramramps** [comment]
Thanks @mjkkirschner  just read the comments.. I am fine with Description...

**Comment by mjkkirschner** [comment]
Any other tests you can think of that I should add here?

File: src/DynamoCore/Graph/Workspaces/SerializationConverters.cs [hunk 3]
```diff
@@ -838,6 +838,8 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
             writer.WriteValue(((TypedParameter)value).Type.rank);
             writer.WritePropertyName("DefaultValue");
             writer.WriteValue(((TypedParameter)value).DefaultValueString);
+            writer.WritePropertyName("Description");
+            writer.WriteValue(((TypedParameter)value).Summary);
```
Comment (ramramps) [inline src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: 👍

**Pull request merged**